File: Facade/c++/Facade.cpp
#include <iostream>

class ProviderCommunication
{
    public:
        void receive();
        void payment();
};

class Site
{
    public:
        void placement();
        void del();
};

class Database
{
    public:
        void insert();
        void del();
};

class MarketPlace
{
    private:
        ProviderCommunication providerCommunication;
        Site site;
        Database database;

    public:
        void productReceipt();
        void productRelease();
};

int main()
{
    MarketPlace marketPlace;

    marketPlace.productReceipt();

    std::cout << std::endl;

    marketPlace.productRelease();
    
    return 0;
}

void ProviderCommunication::receive()
{
    std::cout << "receiving products from the manufacturer\n";
}

void ProviderCommunication::payment()
{
    std::cout << "payment\n";
}

void Site::placement()
{
    std::cout << "placement on the website\n";
}

void Site::del()
{
    std::cout << "removal from the site\n";
}

void Database::insert()
{
    std::cout << "writing to the database\n";
}

void Database::del()
{
    std::cout << "deleting from the database\n";
}

void MarketPlace::productReceipt()
{
    providerCommunication.receive();
    site.placement();
    database.insert();
}

void MarketPlace::productRelease()
{
    providerCommunication.payment();
    site.del();
    database.del();
}

File: Facade/java/Facade.java
class ProviderCommunication{
    public void receive(){
        System.out.println("receiving products from the manufacturer");
    }

    public void payment(){
        System.out.println("payment");
    }
}

class Site{
    public void placement(){
        System.out.println("placement on the website");
    }

    public void del(){
        System.out.println("removal from the site");
    }
}

class Database{
    public void insert(){
        System.out.println("writing to the database");
    }

    public void del(){
        System.out.println("deleting from the database");
    }
}

class MarketPlace{
    private ProviderCommunication providerCommunication;

    private Site site;

    private Database database;
    
    public MarketPlace(){
        providerCommunication = new ProviderCommunication();
        site = new Site();
        database = new Database();
    }

    public void productReceipt(){
        providerCommunication.receive();
        site.placement();
        database.insert();
    }

    public void productRelease(){
        providerCommunication.payment();
        site.del();
        database.del();
    }
}

public class Main{
    public static void main(String[] args){
        MarketPlace marketPlace = new MarketPlace();

        marketPlace.productReceipt();

        System.out.println();

        marketPlace.productRelease();
    }
}